Requirements/cross-chain-transfer.md

```diff
@@ -38,19 +48,55 @@ Similar to the previous case, deposit transactions to the federation's multisig
+If not enough members are online to reach the minimum number of signature required by the multisig, then transfers will halt untill enough members come back online and broadcast signatures.
 
 ![The cross chain is handled one block later by next leader](../assets/cross-chain-transfers/leader-offline-2.svg)
 
 ### The leader comes back online
 
 When the leader comes back online, it will retrieve from its stores what was the last known block heights for which it had dealt with transfers, and resync its store from there while synching up its nodes. Partial transactions will then be added to the store and updated (or potentially completed) depending on their status on the chain B.  
-The node will only be able to participate in cross chain transfers when it is fully synced again, and has access to the latest version of the utxos on each multisigs.
+The node will only be able to participate in cross chain transfers when it is fully synced again, and has the correct view of the multisig utxos set.  
+
+A leader that is now synced and observing unprocessed previous transfers will start creating signatures for those transfers and broadcasting them to the other federation members.
 
 ![The leader comes back online](../assets/cross-chain-transfers/leader-offline-3.svg)
 
 ## 3) Building the chain B transaction deterministically
 
-The main point of building transactions deterministically is to allow for each member to independently build the exact same transaction. This way the can uniquely identify transactions coming from other members and decide to match them with their own (to collect signatures, change their statuses when finding them on chain, or when recovering from a period offline).
+There are two main reasons for building transactions deterministically.
+1. To allow for each member to independently build the exact same transaction, and therefore ensure that the federation members will never need to sign a UTXO more than once. This is for security reasons.
+2. To simplify the processes of sharing signaturs (a none determininstic approach means a leader needs to create a session for the transfer and coordinate with peers to sign it).  
+
+Assuming members are synced and have the same view of the multisig address (the collection of UTXOs) then using a predefined algorithem every member can generate the transfer transaction, sign with their key and broadcast to the other members (possibly its enought to only broadcast to the current leader).  
+It may be that in order to use a multisig UTXO the algorithem will require it to be burried under enough blocks, to avoid reorgs and ensure that all member have the same view of the UTXO list.
+
+**The algorithm to select UTXOs:**
+- The oldest UTXO first
+- If a block has more than one UTXO, then first one in the list is first
+- Outputs are sorted by destination address
+- Change address (back to multisig), if any, is last.
+- OP_RETURN of block hash of chain the transfer came from.
```

Comment:
Again, I presume this is referring to the position of the output in the constructed transaction.